File: 1512.好数对的数目.py
#
# @lc app=leetcode.cn id=1512 lang=python3
#
# [1512] 好数对的数目
#

# @lc code=start
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        m = collections.Counter(nums)
        return sum(v * (v-1) // 2 for k, v in m.items())

# @lc code=end

File: 1512.好数对的数目.cpp
/*
 * @lc app=leetcode.cn id=1512 lang=cpp
 *
 * [1512] 好数对的数目
 */

// @lc code=start
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        unordered_map<int, int>com;
        for(auto ch : nums){
            com[ch] += 1;
        }
        int sum = 0;
        for(auto item : com){
            int count = item.second;
            sum += count*(count-1)/2;
        }
        return sum;
    }
};
// @lc code=end

